### DISCUSSION TITLE
Can we use the Mylife APP application and the xDrip application on the same phone?

### DISCUSSION THREAD

**Discussion by Michel-Cavaud** [NightscoutFoundation/xDrip#2567] [Q&A]
Hello, Can we use the Mylife APP application and the xDrip application on the same phone?
I have a Dexcom 6 sensor and a YpsoPump that I use with xDrip but I would like to have more possibilities with the data from my Gexcom.
I would like to have sensor data on both apps.
Thanks

**Comment by Navid200** [comment]
What does Mylife app do?  If it controls the pump only, there should be no impact on xDrip and Dexcom.

If there is a limitation, it would be by the Dexcom G6 transmitter.  It (G6 transmitter) can be connected to by only one mobile app.  We can talk more about that.  But, it will be unnecessary if the app doesn't connect to G6.

**Comment by Michel-Cavaud** [comment]
Hello,
Thanks for your feedback.
If the Mylife App is connected to the Dexcom G6 pump and sensor.
So given your answer, it is impossible to use another applcation connected to the sensor.
I just wanted to be able to retrieve my sensor data in real time as the Mylife app does not allow this.

Thanks

**Reply by Navid200** [reply]
I don't know anything about the pump you are talking about.

Are you sure it connects to G6?  Why would it?  Is it a basic pump or a closed loop system like t:slim?

If you are sure that it connects to G6, do you know which slot of the G6 transmitter it uses?
A G6 transmitter has two slots.  The t:slim pump uses the medical slot of the transmitter.  This allows xDrip to use the mobile slot at the same time.

Can you provide information about the pump?
Can you at least provide a link?

**Reply by steve8x8** [reply]
If the full name of the app is something like "mylife CamAPS FX" - what I'm suspecting here, the corresponding ID string would be "com.camdiab.fx_alert.mmoll" or "com.camdiab.fx_alert.mgdl" depending on your glucose unit -
- then xDrip cannot connect to the G6 directly because the sensor (transmitter) already talks to the bluetooth address of the phone, providing values for the CamAPS code to control the pump.
- But there's hope: CamAPS is supported by Companion mode right from the beginning. (David Burren wrote about it [here](https://bionicwookiee.com/2022/08/21/mixing-camaps-with-xdrip/).)
- You may also use a BlueJay watch to connect to the sensor, and "upload" the data to xDrip on the phone.

**Reply by jamorham** [reply]
@steve8x8 just to say simultaneous bluetooth is possible on the same device using the `Companion Bluetooth` setting. (Probably needs Android 12 or above to work)

**Reply by steve8x8** [reply]
I'm still at LineageOS 18.1 (Android 11) so I'm of no help with testing/confirming this.

**Reply by Navid200** [reply]
If this is indeed CamAPS, people are using the ability of xDrip to switch to the medical slot while CamAPS uses the incorrect mobile slot.

This however, needs engineering mode to be on.  I wish to remove this capability from the engineering mode so that it can be used in normal mode.  But, before I can do that, I need to add the slot in use to the system status page.
